src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/handlers/TicketHandler.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.ticket.handlers;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalListener;
```

Comment:
Caffeine is much much faster [see the statistics](https://github.com/ben-manes/caffeine/wiki/Benchmarks) and is essentially an improvement upon Guava's cache, I decided to use it in order to improve performance and also because it can better handle this case as it involves futures. It handles caching only completed futures and gracefully handles any errors which may occur during the futures computation.